lib/Image/PNG/Portable.rakumod

```diff
@@ -78,13 +79,34 @@ method write(Str:D $file) {
 
     write-chunk $fh, 'IDAT', compress $!data;
 
+    if %!text-metadata.defined {
```

Comment:
The outer `if` is superfluous.  If `%!text-metadata` is empty, the `for` loop won't run.